### ISSUE TITLE
Please follow the Desktop Entry Specification (OnlyShowIn/NotShowIn)

### ISSUE THREAD

**Issue by jbicha** [linuxmint/cinnamon-session#97]
Issue
====
Please follow the freedesktop Desktop Entry Specification standard.

Cinnamon has been a [registered](https://standards.freedesktop.org/menu-spec/menu-spec-latest.html#onlyshowin-registry) Desktop Environment since [desktop-file-utils](https://cgit.freedesktop.org/xdg/desktop-file-utils) 0.23 which was released over a year ago (and I personally backported this change to Ubuntu 14.04 LTS).

Cinnamon clearly is an independent desktop and it's time for it to [respect](https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys) the `OnlyShowIn` and `NotShowIn` .desktop keys. Please drop your `autostart-blacklist`.

If an app incorrectly excludes Cinnamon, please file bugs for that. In the unlikely case that a patch would be rejected for that upstream, distros will accept your patches.

Example
-----------
Install Cinnamon on top of GNOME and you'll end up with 2 Files apps in your menu. (Do it in Ubuntu and you'll have 3 because of a unique Ubuntu packaging situation). This is silly since Nautilus sets `OnlyShowIn=GNOME;` (and the extra Ubuntu entry also should be hidden appropriately) and Cinnamon has its own file browser so this problem shouldn't exist.

**Comment by leigh123linux** [comment]
It might be a while till this happens due to Mint's LTS base, it is possible to filter the crap from cinnamon menu

```
--- a/files/etc/xdg/menus/cinnamon-applications.menu	2015-07-09 13:17:00.000000000 +0100
+++ b/files/etc/xdg/menus/cinnamon-applications.menu	2015-07-20 10:39:38.975607303 +0100
@@ -31,6 +31,8 @@
 	     applications here -->
         <Not><Category>Accessibility</Category></Not>
         <Not><Category>System</Category></Not>
+        <Not><Filename>org.gnome.Nautilus.desktop</Filename></Not>
+        <Not><Filename>yelp.desktop</Filename></Not>
       </And>
     </Include>
   </Menu> <!-- End Accessories -->
@@ -43,6 +45,7 @@
       <And>
         <Category>Accessibility</Category>
         <Not><Category>Settings</Category></Not>
+        <Not><Filename>onboard.desktop</Filename></Not>
       </And>
     </Include>
   </Menu> <!-- End Accessibility -->
@@ -141,6 +144,9 @@
         <Not><Category>Core</Category></Not>
         <Not><Category>Settings</Category></Not>
         <Not><Category>Screensaver</Category></Not>
+        <Not><Filename>onboard.desktop</Filename></Not>
+        <Not><Filename>redhat-usermount.desktop</Filename></Not>
+        <Not><Filename>fedora-release-notes.webapp.desktop</Filename></Not>
       </And>
     </Include>
   </Menu> <!-- End Other -->
@@ -163,6 +169,11 @@
       <And>
         <Category>Settings</Category>
         <Not><Category>System</Category></Not>
+        <Not><Filename>onboard-settings.desktop</Filename></Not>
+        <Not><Filename>redhat-userinfo.desktop</Filename></Not>
+        <Not><Filename>redhat-userpasswd.desktop</Filename></Not>
+        <Not><Filename>system-config-users.desktop</Filename></Not>
+        <Not><Filename>xscreensaver-properties.desktop</Filename></Not>
       </And>
     </Include>
   </Menu>
@@ -181,6 +192,9 @@
       <And>
         <Category>System</Category>
         <Not><Category>Settings</Category></Not>
+        <Not><Filename>redhat-usermount.desktop</Filename></Not>
+        <Not><Filename>system-config-users.desktop</Filename></Not>
+        <Not><Filename>fedora-release-notes.webapp.desktop</Filename></Not>
       </And>
     </Include>
   </Menu>     <!-- End System Settings -->
```

**Comment by clefebvre** [comment]
I agree with this, but I don't think we're ready for the switch. Incorrect OnlyShowIn cases should first be identified, listed and fixed. I can see many just looking at a default installation in both /usr/share/applications and /etc/xdg/autostart.

Once we're happy with the situation we can consider not starting GNOME apps/services. I've no idea how long this can take but if Debian, Ubuntu and other distros are happy to make the changes it can indeed happen.

While we're at it maybe it's time we also clean up on "X-Cinnamon" vs "Cinnamon"... there's a bit of both here and there.

**Comment by leigh123linux** [comment]
@clefebvre  This shouldn't be an issue for me as I have 'Proven packager' status for Fedora which means I can fix anything broken by this change.


> Once we're happy with the situation we can consider not starting GNOME apps/services. I've no idea how long this can take but if Debian, Ubuntu and other distros are happy to make the changes it can indeed happen.

**Comment by jbicha** [comment]
If you come up with a list of "external" apps and services that need to be adapted, I think it's very achievable to handle them upstream or in distros by the time of Linux Mint's next major release next year.

I use GNOME so I'm not going to prepare the list but I can help fix things on the list. I can push to GNOME (with maintainer consent), Debian unstable/testing and Ubuntu.

If you target Debian 9 "Stretch", you may need to keep the old workaround longer there since it takes longer to fix issues in Debian Stable but since changing the .desktop's is a trivial fix for an important bug, I think it just needs someone motivated enough there.

**Comment by jbicha** [comment]
I assume this bug is still valid so it should not be auto-closed.
